src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -432,18 +432,26 @@ public void Invoke(ScriptInvocationContext context)
 
         internal void InvokeResponse(InvocationResponse invokeResponse)
         {
-            if (_executingInvocations.TryRemove(invokeResponse.InvocationId, out ScriptInvocationContext context)
-                && invokeResponse.Result.IsSuccess(context.ResultSource))
+            if (_executingInvocations.TryRemove(invokeResponse.InvocationId, out ScriptInvocationContext context))
             {
-                IDictionary<string, object> bindingsDictionary = invokeResponse.OutputData
+                if (invokeResponse.Result.IsSuccess(context.ResultSource))
+                {
+                    IDictionary<string, object> bindingsDictionary = invokeResponse.OutputData
                     .ToDictionary(binding => binding.Name, binding => binding.Data.ToObject());
 
-                var result = new ScriptInvocationResult()
+                    var result = new ScriptInvocationResult()
+                    {
+                        Outputs = bindingsDictionary,
+                        Return = invokeResponse?.ReturnValue?.ToObject()
+                    };
+                    context.ResultSource.SetResult(result);
+                }
+                else
                 {
-                    Outputs = bindingsDictionary,
-                    Return = invokeResponse?.ReturnValue?.ToObject()
-                };
-                context.ResultSource.SetResult(result);
+                    Exception invocationException = invokeResponse.Result.InvocationResultException();
+                    context.Logger.LogError(invocationException.Message);
```

Comment:
@brettsam - Any ideas if this would cause duplicate logs in AppInsights as this change explicity logs errors to function logger

Reply:
Yes, it would be good to verify that we're not logging twice to app insights

Reply:
If you have a failed function invocation with an exception, it should be "naturally" logged when it throws back to the FunctionExecutor. I'm pretty sure this line isn't needed. We shouldn't be logging anything with the function logger. 

When you set the exception below, does that cause InvokeCore() to throw an exception?

Reply:
cc: @paulbatum , @davidebbo 
I just tried the same with a c# function. It is the same behavior. We do not log invocation exceptions to file logger. Perhaps by design?
For c#, compilation exceptions are logged, but for any out-of-proc language we only get errors on invocation.